### PR TITLE
Use SystemConfig instead of AllConfig for DB stuff

### PR DESCRIPTION
**Pull request by MorrisJobke**
* preparation for followup PRs to clean up the DB bootstrapping

@nickvergessen @rullzer @blizzz @LukasReschke This is basically a split off from another bigger PR and the other one would just have been too big to useful review it. So this is the first version of a series of cleanups that are needed to downstream some bigger PRs.

### CODE REVIEW

File: autotest.sh [hunk 1]
```diff
@@ -143,6 +143,8 @@ function cleanup_config {
 	if [ -f config/autotest-storage-swift.config.php ]; then
 		rm config/autotest-storage-swift.config.php
 	fi
+	# Remove mysqlmb4.config.php
+	rm -f config/mysqlmb4.config.php
```
Comment (rullzer) [inline autotest.sh]: Is this part of the PR?

**Comment by rullzer** [comment]
I like it.

but failing tests.

File: autotest.sh (hunk 1 above)
Comment (MorrisJobke) [inline autotest.sh]: I put it in, because minor stuff, but let me look into splitting it out and check out the tests.

**Comment by MorrisJobke** [comment]
> but failing tests.

I forgot to update the mocks of one test class 🙈 Should work now

File: autotest.sh (hunk 1 above)
Comment (MorrisJobke) [inline autotest.sh]: Moved to ##3927

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**